spock-core/src/main/java/org/spockframework/runtime/FeatureNode.java

```diff
@@ -42,8 +43,25 @@ public void around(SpockExecutionContext context, Invocation<SpockExecutionConte
     errorInfoCollector.assertEmpty();
   }
 
+  @Override
+  public SpockExecutionContext prepare(SpockExecutionContext context) throws Exception {
+    alreadyPrepared = true;
+    return super.prepare(context);
+  }
+
+  @Override
+  public SpockExecutionContext execute(SpockExecutionContext context, DynamicTestExecutor dynamicTestExecutor) throws Exception {
```

Comment:
Running other things in `execute` is awkward as it will have already initialized the current test instance and executed its `setup`. This can lead to all kind of weird problems, e.g. `@Shared int counter` would show weird behavior if incremented in the `setup`.

Furthermore, I think this will cause issues with `@ResourceLock`, trying it threw this error.

```
org.junit.platform.commons.JUnitException: Dynamic test descriptors must not declare exclusive resources: [ExclusiveResource [key = 'foo', lockMode = READ_WRITE]]
```

Reply:
Hm, bad about the resource lock indeed, while usage wise the resource lock could of course be acquired on the spec as it has to be `SAME_THREAD` anyway as mentioned above.

Modifying shared fields in `setup` is indeed a bit unlucky.
Unfortunately, the `DynamicTestExecutor` (which was also Marcs first idea) is only available in the `execute` method. :-/
But due to the resource lock problem it might not be the best idea anyway.